### PR TITLE
Set default value for half_block = 0 when reading ROOT files

### PR DESCRIPTION
**Pull request by robbietuk**
Fix for #1008?

### CODE REVIEW

**Comment by KrisThielemans** [comment]
I suggest to have an option for compatibility with STIR 4, setting it to the old version. We should have given people notice about this change, so this way they can at least keep the same results. Obviously I'd default it to off (i.e. `half_block=0`). Not sure how to call it. feel free to suggest something

**Comment by robbietuk** [comment]
Yes. However, it still needs a test because now I am concerned about rotations.

**Comment by KrisThielemans** [comment]
only now? 😄

File: CMakeLists.txt [hunk 1]
```diff
@@ -141,6 +141,7 @@ endif()
 # Legacy options
 option(STIR_PROJECTORS_AS_V3 "Enable STIR version 3 legacy code" OFF)
 option(STIR_LEGACY_IGNORE_VIEW_OFFSET "Ignore using scanner view-offset (or intrinsic azimuthal tilt), as in STIR 4 or earlier" OFF)
+option(STIR_ROOT_AS_V4 "Ignore the changes to the ROOT rotation introduced in STIR version 5" OFF)
```
Comment (robbietuk) [inline CMakeLists.txt:144]: Could do `STIR_ROOT_ROTATION_AS_V4`?

File: src/IO/InputStreamFromROOTFileForCylindricalPET.cxx [hunk 2]
```diff
@@ -146,6 +148,13 @@ InputStreamFromROOTFileForCylindricalPET::set_defaults()
     module_repeater_y = -1;
     module_repeater_z = -1;
     rsector_repeater = -1;
+#if STIR_ROOT_AS_V4
+    half_block = module_repeater_y * submodule_repeater_y * crystal_repeater_y / 2  - 1;
+    if (half_block < 0 )
+      half_block = 0;
+#else
+    half_block = 0;
+#endif
```
Comment (robbietuk) [inline src/IO/InputStreamFromROOTFileForCylindricalPET.cxx:157]: Keeping `half_block` for now. Can be removed when `STIR_ROOT_AS_V4=false` later.

**Review by robbietuk — COMMENTED** [review]
Forgot to commit before a forcepush... Apparently that closes the PR, who knew.

File: CMakeLists.txt (hunk 1 above)
Comment (KrisThielemans) [reply CMakeLists.txt:144]: Ok for me

**Comment by robbietuk** [comment]
Changed `STIR_ROOT_AS_V4` to `STIR_ROOT_ROTATION_AS_V4` as it will be more useful later. I think this is ready to go and we can test later? It will likely mean that a number of older GATE geometries are now rotated in version5 but we are going to fix that later?

File: documentation/release_5.0.htm [hunk 3]
```diff
@@ -339,6 +339,11 @@ <h3>Backward incompatibities</h3>
       scanners, this will fill in the detector maps etc.
       You will get a run-time exception if you forgot to do this.
     </li>
+    <li>
+        Rotational changes to STIR's interface with ROOT files as subtraction of a <code>half_block</code> has been remove the from the determination of the crystalIDs.
```
Comment (KrisThielemans) [inline documentation/release_5.0.htm:343]:
```suggestion
        Rotational changes to STIR's interface to ROOT files. In previous versions (roughly) half a bucket,
        <code>(module_repeater_y * submodule_repeater_y * crystal_repeater_y) /2 - 1)</code>
        in the <tt>Cylindrical</tt> case, was subtracted from the crystalIDs, in an attempt to rotate
        the scanner. However, this was incompatible with the STIR code to find normalisation factors,
        and could be surprising for odd numbers etc.
```

**Pull request merged**